src/tools/msc.lua

```diff
@@ -237,13 +246,21 @@
 -- Decorate include file search paths for the MSVC command line.
 --
 
-	function msc.getincludedirs(cfg, dirs, sysdirs, frameworkdirs)
+	function msc.getincludedirs(cfg, dirs, sysdirs, extdirs, frameworkdirs)
 		local result = {}
 		dirs = table.join(dirs, sysdirs)
 		for _, dir in ipairs(dirs) do
 			dir = project.getrelative(cfg.project, dir)
 			table.insert(result, '-I' ..  p.quoted(dir))
 		end
+		for _, dir in ipairs(extdirs or {}) do
+			dir = project.getrelative(cfg.project, dir)
+			if _ACTION >= "vs2022" then
```

Comment:
Logically, test should be against the toolset version, as other actions can use msc toolset.

Reply:
I was thinking that as well. How do I check that here?

Reply:
I would say something like `cfg.toolset >= "msc-v143"`.

Reply:
@starkos: from [toolset](https://premake.github.io/docs/toolset/), it seems we can use `"v143"` directly instead of `"msc-v143"`, I didn't see code to unify it afterward.
As `"vyyy" > "msc-xxx"`, that check might be problematic. which is the correct method?

Reply:
`cfg.toolset >= "msc-v143"` seems the way to go.

Reply:
Changing this line to `if cfg.toolset >= "msc-v143" then` causes an error in the tests:

```
[  FAILED  ] 1 tests, listed below:
[  FAILED  ] tools_msc.cflags_onExternalIncludeDirs
[string "src/tools/msc.lua"]:187: attempt to compare string with nil
stack traceback:
        [string "src/tools/msc.lua"]:187: in field 'getincludedirs'
        ...s/engle/source/oss/premake-core/tests/tools/test_msc.lua:301: in function 'testFunction'
```

Seems like the `cfg.toolset` value is not always set.

Reply:
`cfg.toolset and cfg.toolset >= "msc-v143"` here and set toolset or action in test_msc.lua does the job.
Actually, in UT of toolset, we don't set neither action nor toolset, that'is why you got issue.

Reply:
Fixed.